packages/apollo-codegen/src/generate.ts

```diff
@@ -58,36 +58,34 @@ export default function generate(
   }
   else if (target === 'flow-modern' || target === 'typescript-modern' || target === 'ts-modern') {
     const context = compileToIR(schema, document, options);
-    const generatedFiles = target === 'flow-modern'
+    const { generatedFiles, common } = target === 'flow-modern'
       ? generateFlowModernSource(context)
       : generateTypescriptModernSource(context) ;
 
-    // Group by output directory
-    const filesByOutputDirectory: {
-      [outputDirectory: string]: {
-        [fileName: string]: BasicGeneratedFile
-      }
+    const outFiles: {
+      [fileName: string]: BasicGeneratedFile
     } = {};
 
-    Object.keys(generatedFiles)
-      .forEach((filePath: string) => {
-        const outputDirectory = path.dirname(filePath);
-        if (!filesByOutputDirectory[outputDirectory]) {
-          filesByOutputDirectory[outputDirectory] = {
-            [path.basename(filePath)]: generatedFiles[filePath]
-          };
-        } else {
-          filesByOutputDirectory[outputDirectory][path.basename(filePath)] = generatedFiles[filePath];
-        }
-      })
-
-    Object.keys(filesByOutputDirectory)
-      .forEach((outputDirectory) => {
-        writeGeneratedFiles(
-          filesByOutputDirectory[outputDirectory],
-          outputDirectory
-        );
-      });
+    const outputIndividualFiles = fs.existsSync(outputPath) && fs.statSync(outputPath).isDirectory();
```

Comment:
Awesome! I think we should make sure to document that folder outputs are accepted in addition to having single files